### ISSUE TITLE
Failed to estimate gas

### ISSUE THREAD

**Issue by kacper-koza-arianelabs** [hashgraph/hedera-mirror-node#7368]
### Description

 I have an issue while trying to estimate gas for a contract call.  This is the call that I want to estimate gas for: 
```
  function getNFT(GetNFTParams memory params) external payable {
        // We know the length of the array
        uint validationGatesCount = getConfig().validationGates.length;
        for (uint i = 0; i < validationGatesCount; i++) {
            (bool success, bytes memory data) = getConfig()
                .validationGates[i]
                .delegatecall(
                    abi.encodeWithSelector(
                        IValidationGate.validate.selector,
                        params
                    )
                );
            require(success, "A validation was not successfully called");
            require(
                abi.decode(data, (bool)),
                "A validation was not successful"
            );

        }

        processPayment(params);

        int64[] memory serialNumber = new int64[](1);
        (bool success, bytes memory data) = getConfig()
            .serialNumberGenerator
            .delegatecall(
                abi.encodeWithSelector(
                    ISerialNumberGeneratorModule.generateSerialNumber.selector
                )
            );

        require(
            success,
            "A ISerialNumberGeneratorModule was not successfully called"
        );

        serialNumber[0] = abi.decode(data, (int64));
        s.nftsDistributed = s.nftsDistributed + 1;
        s.leftToDistribute = s.leftToDistribute - 1;
        transferNFTsPublic(msg.sender, serialNumber);

        for (uint i = 0; i < validationGatesCount; i++) {
            (bool success, bytes memory data) = getConfig()
                .validationGates[i]
                .delegatecall(
                    abi.encodeWithSelector(
                        IValidationGate.hookAfterDistributed.selector,
                        params
                    )
                );
            require(
                success,
                "A hookAfterDistributed was not successfully called"
            );
        }

        (
            bool successAfterDistributed,
            bytes memory dataAfterDistributed
        ) = getConfig().serialNumberGenerator.delegatecall(
                abi.encodeWithSelector(
                    ISerialNumberGeneratorModule.hookAfterDistributed.selector,
                    serialNumber[0]
                )
            );
        require(
            successAfterDistributed,
            "A ISerialNumberGeneratorModule.hookAfterDistributed was not successfully called"
        );

        if(!s.firstClaimed) {
            s.firstClaimed = true;
        }
    }
```
This method works fine, I can use it in my app without errors. The problem occurs when I try to estimate gas for this call. 
Error that I received: 

```
estimateGas.ts:39 Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="[Request ID: 1a4eedb6-dbff-482c-a88b-3e8b02417d8d] execution reverted: ", method="estimateGas", transaction={"to":"0x00000000000000000000000000000000006717A8","data":"0xe88a64390000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"error\":{\"code\":-32008,\"name\":\"Contract revert executed\",\"message\":\"[Request ID: 1a4eedb6-dbff-482c-a88b-3e8b02417d8d] execution reverted: \",\"data\":\"0x\"},\"jsonrpc\":\"2.0\",\"id\":44}","error":{"code":-32008,"data":"0x"},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"to\":\"0x00000000000000000000000000000000006717a8\",\"data\":\"0xe88a64390000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"}],\"id\":44,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://testnet.hashio.io/api"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)
    at _Logger.makeError (chunk-GY3CDWMH.js?v=a65d3185:2291:19)
    at _Logger.throwError (chunk-GY3CDWMH.js?v=a65d3185:2300:16)
    at checkError (ethers.js?v=a65d3185:11375:16)
    at JsonRpcProvider.<anonymous> (ethers.js?v=a65d3185:11857:16)
    at Generator.throw (<anonymous>)
    at rejected (ethers.js?v=a65d3185:11316:32)
overrideMethod @ console.js:213
estimateGas @ estimateGas.ts:39
await in estimateGas (async)
(anonymous) @ HDC.ts:104
handleSDKError @ handleHashgraphSdkError.ts:5
createGetNFTTransaction @ HDC.ts:81
createClaimTransaction @ HDC.ts:129
(anonymous) @ useClaiming.ts:145
onClick @ ContractClaimingButton.tsx:35
callCallback2 @ react-dom_client.js?v=a65d3185:13208
invokeGuardedCallbackDev @ react-dom_client.js?v=a65d3185:13233
invokeGuardedCallback @ react-dom_client.js?v=a65d3185:13267
invokeGuardedCallbackAndCatchFirstError @ react-dom_client.js?v=a65d3185:13270
executeDispatch @ react-dom_client.js?v=a65d3185:16550
processDispatchQueueItemsInOrder @ react-dom_client.js?v=a65d3185:16570
processDispatchQueue @ react-dom_client.js?v=a65d3185:16579
dispatchEventsForPlugins @ react-dom_client.js?v=a65d3185:16587
(anonymous) @ react-dom_client.js?v=a65d3185:16711
batchedUpdates$1 @ react-dom_client.js?v=a65d3185:28443
batchedUpdates @ react-dom_client.js?v=a65d3185:13113
dispatchEventForPluginEventSystem @ react-dom_client.js?v=a65d3185:16710
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom_client.js?v=a65d3185:15012
dispatchEvent @ react-dom_client.js?v=a65d3185:15006
dispatchDiscreteEvent @ react-dom_client.js?v=a65d3185:14983
```

This is my node function for gas estimation

```
export const estimateGas = async ({
  network,
  contractId,
  contractAbi,
  method,
  callArguments
}: {
  network: string;
  contractId: ContractId;
  contractAbi: ethers.ContractInterface;
  method: string;
  callArguments: unknown[];
}): Promise<number | null> => {
  let gas: number | null = null;

  try {
    const provider = new ethers.providers.JsonRpcProvider(`https://${network}.hashio.io/api`);
    const myContract = new Contract(contractId.toSolidityAddress(), contractAbi);
    const populateTransaction = myContract?.populateTransaction?.[method];

    if (populateTransaction) {
      const txToEstimate = await populateTransaction(...callArguments);
      const gasEstimate = await provider.estimateGas(txToEstimate);
      gas = gasEstimate.toNumber();
    }
  } catch {
    return null;
  }

  return gas;
};
```
I have attached the solidity code for this smart contract.

[SmartContracts.zip](https://github.com/hashgraph/hedera-mirror-node/files/13647258/SmartContracts.zip)

### Steps to reproduce

I have attached zip file with code that you can run to see problem. You have there hardcoded contract id. 

Use with node

`npm i`
`node index.js`

[failed-estimation.zip](https://github.com/hashgraph/hedera-mirror-node/files/13648301/failed-estimation.zip)

### Hedera network

testnet

### Version

node v18.18.1

### Operating system

Windows

**Comment by 0xivanov** [comment]
try this:

```js
import { Contract, ethers } from "ethers"; // 5.7.2
import DistributionContractAbi from './DistributionContractAbi.json' assert { type: "json" };
import { ContractId } from "@hashgraph/sdk";  // 2.36.0

async function main() {
  const gas = await estimateGas({
    network: "testnet",
    contractId: ContractId.fromString("0.0.6763923"),
    contractAbi: DistributionContractAbi.abi,
    method: "getNFT",
    callArguments: [{ merkleProof: [], paymentToken: "0x0000000000000000000000000000000000000000" }],
  });

  console.log(gas)
}

main()



async function estimateGas({
  network,
  contractId,
  contractAbi,
  method,
  callArguments
}) {
  let gas = null;

  try {
    const provider = new ethers.providers.JsonRpcProvider(`https://${network}.hashio.io/api`);
    const myContract = new Contract(contractId.toSolidityAddress(), contractAbi);
    const populateTransaction = myContract?.populateTransaction?.[method];

    if (populateTransaction) {
      const txToEstimate = await populateTransaction(...callArguments);
      txToEstimate.from = "0x000000000000000000000000000000000022418f"
      const gasEstimate = await provider.estimateGas(txToEstimate);
      gas = gasEstimate.toNumber();
    }
  } catch (e) {
    console.error(e)
    return null;
  }

  return gas;
};
```

**Comment by kacper-koza-arianelabs** [comment]
@bilyana-gospodinova I updated the bug description. Estimation doesn't work for the `initialize` method. I also saw that in this example https://gitpod.io/new/#https://github.com/ed-marquez/hedera-example-eth-call-estimategas-hip584 signer was passed to `new Contract(contractId.toSolidityAddress(), contractAbi, signer)`. In my example, we don't create a signer but I tested  solution with a signer locally: 
```
const signer = new ethers.Wallet(
      PrivateKey.fromStringDer(
        '302e02...........'
      ).toStringRaw(),
      provider
    );
    const myContract = new Contract(contractId.toSolidityAddress(), contractAbi, signer);
```
I faced the same errror
